.github/workflows/audit-check.yml

```diff
@@ -3,12 +3,51 @@ name: Security audit
 on:
   schedule:
     - cron: "0 0 * * *"
+  pull_request:
+    branches: [ main ]
 
 jobs:
   audit:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      issues: write
+      checks: write
     steps:
       - uses: actions/checkout@v5
-      - uses: actions-rs/audit-check@v1
+      - uses: rustsec/audit-check@v2.0.0
```

Comment:
this is the maintained, not-deprecated alternative ^^

https://github.com/rustsec/audit-check